bin/templates/cordova/lib/Java.js

```diff
@@ -0,0 +1,157 @@
+        let version = null;
+        try {
+            version = (await execa('javac', ['-version'], { all: true })).all;
+        } catch (ex) {
+            events.emit('verbose', ex.shortMessage);
+
+            let msg =
+            'Failed to run "javac -version", make sure that you have a JDK version 8 installed.\n' +
+            'You can get it from the following location:\n' +
+            'https://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html';
+            if (process.env.JAVA_HOME) {
+                msg += '\n\n';
+                msg += 'Your JAVA_HOME is invalid: ' + process.env.JAVA_HOME;
+            }
+            throw new CordovaError(msg);
+        }
+
+        return Java._parseVersion(version);
+    }
+
+    /**
+     * Parses a java version output, ie: "javac 1.8.0_275"
+     *
+     * @param {String} version
+     * @returns {Object}
+     * {
+     *      major: number
+     *      minor: number
+     *      patch: number
+     *      build: number
+     * }
+     */
+    static _parseVersion (version) {
+        version = version.replace('javac ', '');
+
+        const mainParts = version.split('.');
+
+        const output = {
+            major: parseInt(mainParts[0]),
+            minor: parseInt(mainParts[1]),
+            patch: null,
+            build: null,
+            string: version
+        };
+
+        const buildParts = mainParts[2].split('_');
+        output.patch = parseInt(buildParts[0]);
+        output.build = parseInt(buildParts[1]);
+
+        return output;
+    }
+
+    /**
+     * Ensures that Java is installed. Will throw exception if not.
+     * Will set JAVA_HOME and PATH environment variables.
+     *
+     * This function becomes a no-op if already ran previously.
+     */
+    static async _ensure () {
+        if (javaIsEnsured) {
+            return;
+        }
+
+        const javacPath = utils.forgivingWhichSync('javac');
+        const hasJavaHome = !!process.env.JAVA_HOME;
+        if (hasJavaHome) {
+            // Windows java installer doesn't add javac to PATH, nor set JAVA_HOME (ugh).
+            if (!javacPath) {
+                process.env.PATH += path.delimiter + path.join(process.env.JAVA_HOME, 'bin');
+            }
+        } else {
+            if (javacPath) {
+                // OS X has a command for finding JAVA_HOME.
+                const find_java = '/usr/libexec/java_home';
+                const default_java_error_msg = 'Failed to find \'JAVA_HOME\' environment variable. Try setting it manually.';
+                if (fs.existsSync(find_java)) {
+                    try {
+                        process.env.JAVA_HOME = (await execa(find_java)).stdout;
+                        console.log('JAVA HOME', process.env.JAVA_HOME);
+                    } catch (ex) {
+                        if (ex) {
+                            events.emit('verbose', ex.shortMessage);
+                        }
```

Comment:
I know that the original version had this if-clause too, but according to the `execa` docs, `ex` should always be a `childProcessResult` and thus truthy. So I would suggest to use 
```suggestion
                        events.emit('verbose', ex.shortMessage);
```